docs/spec_draft.md

```diff
@@ -1541,6 +1547,71 @@ operations correspond to [stablehlo.minimum](#stablehlominimum) and
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.collective_permute
+
+### Semantics
+
+Within each process group in the StableHLO grid, sends the value of the
+`operand` tensor from the source process to the target process and produces a
+`result` tensor.
+
+The operation splits the StableHLO grid into `process_groups` as follows:
+
+  * `channel_id <= 0`,
+    `cross_replica(replica_groups)`.
+  * `channel_id > 0`,
+    `cross_partition(replica_groups)`.
+
+Afterwards, `result@process` is given by:
+
+ * `operand@process_groups[i, 0]`, if there exists an `i` such that `process_groups[i, 1] = process`.
+ * `[0, ..., 0]`, otherwise.
+ * `broadcast_in_dim(0, [], shape(result))`, otherwise.
+
+### Inputs
+
+| Name                  | Type                                         |
+|-----------------------|----------------------------------------------|
+| `operand`             | tensor of any supported type                 |
+| `source_target_pairs` | 2-dimensional tensor constant of type `si64` |
+| `channel_id`          | constant of type `si64`                      |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) dim(`source_target_pairs`, 1) $=$ 2.
+  * (C2) All values in `source_target_pairs[:, 0]` are unique.
+  * (C3) All values in `source_target_pairs[:, 1]` are unique.
+  * (C4) $0 \ge$ source_target_pairs[i][0], source_target_pairs[i][1] $\lt N$,
+         where $N$ is given by
```

Comment:
"depends on the process grouping strategy" to be consistent with AllGatherOp. Otherwise, it's unclear what "if cross_replica" means.